### PR TITLE
Liberty

### PR DESCRIPTION
**Pull request by pjbreaux**
@mattgreene 

#### What's this change do?
This PR is to merge liberty into master. Once that is done, I will create a branch for liberty and mitaka based on the tip of master. Both liberty and mitaka installations of these playbooks have been tested.

#### Where should the reviewer start?
Review the license, contributing doc, and README file, just to make sure it's fit for public consumption.

### CODE REVIEW

**Comment by pjbreaux** [comment]
All yaml files have been reviewed by @dflanigan, but feel free to peruse them if you'd like

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<!--
+Copyright 2016 F5 Networks Inc.
```
Comment (mattgreene) [inline CONTRIBUTING.md]: Fix copyright to include 2017.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Creating issues is good, creating good issues is even better. Filing meaningful bug reports with lots of information in them helps us figure out what to fix when and how it impacts our users. We like bugs because it means people are using our code, and we like fixing them even more.
+ 
+## Pull Requests
+If you are submitting a pull request, you need to make sure that you have done a few things first.
+
+* If an issue doesn't exist, file one.
+* Make sure you have tested your code because we are going to do that when when you make your PR.  You don't want _The Hat_ because your request fails unit tests.
+*  Clean up your git history because no one wants to see 75 commits for one issue
+*  Use our [commit template](.git-commit-template.txt)
+*  Use our pull request template
```
Comment (mattgreene) [inline CONTRIBUTING.md:30]: Create the magic .github directory with the following files:
PULL_REQUEST_TEMPLATE.md
ISSUE_TEMPLATE.md

See the agent repo for samples.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
+F5 OpenStack Ansible
+====================
+
+|slack badge|
```
Comment (mattgreene) [inline README.rst:4]: I see the new ansible channel on F5OpenStack slack.  Are you also going to create a channel on F5OpenStackDev slack?

File: playbooks/hosts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[controllers]
+10.190.4.147 ansible_ssh_private_key_file=/root/.ssh/id_rsa_testlab ansible_ssh_extra_args='-o StrictHostKeyChecking=no'
```
Comment (mattgreene) [inline playbooks/hosts]: Remove the real IP address so someone doesn't accidentally clober this machine in the future.

File: playbooks/roles/configure_lbaasv2_agent/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# version: 2016-12-20
+---
```
Comment (mattgreene) [inline playbooks/roles/configure_lbaasv2_agent/tasks/main.yml:5]: Can you add a comment to this file simply to state that the agent ini contains many settings, but this file is starting with the most commonly used settings?  (and feel free to add more as required)

File: playbooks/roles/configure_lbaasv2_agent/vars/main.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+---
+undercloud: true
```
Comment (mattgreene) [inline playbooks/roles/configure_lbaasv2_agent/vars/main.yml]: We've been conflating over and under cloud with GRM or L2 Adjacent Mode.  Rename this with "f5_global_routed_mode" which can be use directly to populate the ini file.

File: playbooks/vars/agent_configuration_vars.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Enter the values for the agent configuration
+bigip_netloc: <bigip_ip_address_or_hostname>
+agent_ini_file: /etc/neutron/services/f5/f5-openstack-agent.ini
+undercloud: <true_or_false>     # Set to true by default. Set to false if using an undercloud deployment.
```
Comment (mattgreene) [inline playbooks/vars/agent_configuration_vars.yaml]: See my previous comment in playbooks/roles/configure_lbaasv2_agent/vars/main.yml.  In fact, is the main.yml file necessary if this setting exists here?

File: playbooks/vars/agent_configuration_vars.yaml [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+tunnel_types: <tunnel_type>     # vxlan, gre etc...
```
Comment (mattgreene) [inline playbooks/vars/agent_configuration_vars.yaml]: Rename to "advertised_tunnel_types" to match the same name in the ini file.

File: playbooks/agent_driver_deploy.yaml [hunk 9]
```diff
@@ -0,0 +1,44 @@
+---
+# version: 2016-12-29
+- hosts: all
+  vars_files:
+    - vars/agent_driver_deploy_vars.yaml
+  become_user: root
+  become_method: sudo
+  become: true
+  remote_user: '{{ remote_user }}'
+  roles:
```
Comment (mattgreene) [inline playbooks/agent_driver_deploy.yaml:10]: We need to be able to locate the agent on a different host than the driver (neutron server).  Can we either split driver and agent apart into separate files, or somehow provide the 'roles' on a per host basis?

**Review by mattgreene — CHANGES_REQUESTED** [review]

File: README.rst (hunk 3 above)
Comment (pjbreaux) [reply README.rst:4]: Done

File: playbooks/vars/agent_configuration_vars.yaml (hunk 7 above)
Comment (pjbreaux) [reply playbooks/vars/agent_configuration_vars.yaml]: We can remove the file and require the user to set the value

File: playbooks/agent_driver_deploy.yaml (hunk 9 above)
Comment (pjbreaux) [reply playbooks/agent_driver_deploy.yaml:10]: This just sounds like a different playbook. Would you like me to add a new one?

Comment (mattgreene) [reply playbooks/agent_driver_deploy.yaml:10]: I prefer separate playbooks for driver and agent.

File: playbooks/vars/agent_configuration_vars.yaml (hunk 7 above)
Comment (mattgreene) [reply playbooks/vars/agent_configuration_vars.yaml]: ok

File: playbooks/agent_driver_deploy.yaml (hunk 9 above)
Comment (pjbreaux) [reply playbooks/agent_driver_deploy.yaml:10]: Updating to add new playbooks for driver deployment, then a separate one for the agent deployment.

**Comment by pjbreaux** [comment]
@mattgreene: I addressed your comments. I created two new playbooks. The deployment of the agent alone and of the driver alone. I kept the original playbook to deploy both as well. For each playbook, I add a vars file. One vars file would be more confusing to just deploy the agent, but we could do that, if you'd like.

**Comment by mattgreene** [comment]
This code is fine for 'master', but when it splits into liberty and mitaka branches the vars file needs to reflect the correct package versions.  playbooks/vars/agent_deploy_vars.yaml refers to 9.1.0 (which is mitaka, though your PR origin branch is liberty).

**Comment by pjbreaux** [comment]
Closing

**Pull request merged**